liquibase/README.md

```diff
@@ -0,0 +1,340 @@
+# Creating Liquibase snapshots for OpenMRS
+This document explains why Liquibase snapshots are introduced to OpenMRS and how versioning of database changes is 
+affected by snapshots. It also contains instructions for creating new Liquibase snapshots, running them from the 
+console and adding them to the OpenMRS code base.
+
+## Why Liquibase snapshots?
+OpenMRS uses [Liquibase](http://www.liquibase.org/index.html) to evolve its database model. The respective change sets 
+have grown in the course of time and contain unneeded changes. E.g. there are tables or columns that are created, 
+modified and eventually deleted as they are no longer used.
+
+Large and partly outdated Liquibase change sets affect OpenMRS in two ways:
+
+* Unneeded changes slow down the initialisation of OpenMRS
+* Liquibase change sets are hard to read and understand as they contain unused code.
+
+An alternative model is based on snapshots where historic change sets are consolidated into smaller change sets.
+
+## Folder structure and naming conventions
+This section describes the folder structure and naming conventions used for Liquibase snapshots and updates.
+
+**Before** the introduction of snapshots, all Liquibase change log files were located in the resources folder of `openmrs-api`:
+
+* `openmrs-core/api/src/main/resources`
+
+**Since** the introduction of snapshots, the respective change log files are no longer stored in the resource folder but in the following subfolders:
+
+*  `openmrs-core/api/src/main/resources/org/openmrs/liquibase/snapshots` contains all Liquibase snapshot files.
+
+*  `openmrs-core/api/src/main/resources/org/openmrs/liquibase/updates` contains all Liquibase update files.
+
+Version numbers are part of the change log file name: 
+
+* `liquibase-update-to-latest-2.1.x.xml`.
+
+## Versioning of database changes with snapshots
+This section compares Liquibase files before and after the introduction of Liquibase snapshots.
+
+### Change sets in [OpenMRS 2.1.x](https://github.com/openmrs/openmrs-core/tree/2.1.x/api/src/main/resources)
+OpenMRS 2.1.x is the **last version that contains the full history** of database changes. The respective Liquibase 
+files are:
+
+* `org/openmrs/liquibase/snapshots/schema-only/liquibase-schema-only-1.9.x.xml` defines the OpenMRS schema
```

Comment:
Could you also remind me why we chose `1.9.x` instead of `1.10.x` or `1.11.x` or `1.12.x`?

Reply:
That was a similar decision, at the time 1.9.x was the starting point. Today we would probably start with 2.4.x.